### PR TITLE
Add `retext-case-police` to list of plugins

### PR DESCRIPTION
**Pull request by JulianCataldo**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

[Repository](https://github.com/JulianCataldo/retext-case-police)

A `retext` plugin for checking popular names casing.

Examples:

- ⚠️ `macbook` → ✅ **`MacBook`**
- ⚠️ `MacOS` → ✅ **`macOS`**

For:

- [Abbreviates](https://github.com/antfu/case-police/blob/main/dict/abbreviates.json)
- [Brands](https://github.com/antfu/case-police/blob/main/dict/brands.json)
- [General](https://github.com/antfu/case-police/blob/main/dict/general.json)
- [Products](https://github.com/antfu/case-police/blob/main/dict/products.json)
- [Softwares](https://github.com/antfu/case-police/blob/main/dict/softwares.json)

---

Dictionaries are from [`case-police`](https://github.com/antfu/case-police).

### CODE REVIEW

**Comment by wooorm** [comment]
Cool, looks useful!

- I recommend a message of the form `` Expected casing of `${correctOne}` instead of `${original}` ``
  - Puts what’s expected in front
  - Uses markdown (code) for “literal” values
- I recommend adding `actual`, `expected`, and `url` fields to messages.
  - `actual` is supposed to hold the actual value in the document: `toString(nodes)`
  - `expected` is supposed to hold the suggestion (an array of strings or one string): `correctOne`
  - `url` can go to documentation, preferrably specific to the `ruleId`
- I’m not sure why you’re checking `if (nodes[0].type === 'WordNode') {`? I don’t think that’s every `false`?

File: doc/plugins.md [hunk 1]
```diff
@@ -73,7 +75,7 @@ The list of plugins:
 *   [`retext-syntax-mentions`](https://github.com/retextjs/retext-syntax-mentions)
     — classify `@mentions` as syntax
 *   [`retext-syntax-urls`](https://github.com/retextjs/retext-syntax-urls)
-    — Classify url-like values (example.com, example.md, etc) as syntax
+    — classify url-like values (example.com, example.md, etc) as syntax
```
Comment (wooorm) [inline doc/plugins.md:78]: good catch :)

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
- You might also want to support `options.ignore`, which similar plugins supports too. E.g., https://github.com/retextjs/retext-passive#optionsignore
- `case-police` seems to do some fancy things on which casings they include and which they ignore: https://github.com/antfu/case-police/blob/6cfa330a55e987038d1a618cb163ebadfa8fbbae/src/utils.ts#L46-L54. Maybe good to add some tests for that?

**Comment by JulianCataldo** [comment]
> * I’m not sure why you’re checking if (nodes[0].type === 'WordNode') {? I don’t think that’s every false?

Screenshot 2022-08-31 at 23 29 20

Good question.
I've added it after trial and error, for making ESLint happy 🤔 Even if it feels redundant. Smash that squiggles 😄

**Comment by wooorm** [comment]
This can likely be solved by using `nodes[0]` as the place, instead of delving into it.

A word like `don’t`, consists of `text("don"), punctuation("’"), text("t")`. Right now you are warning about `don`, but the warning is instead about the whole word?

**Comment by JulianCataldo** [comment]
Oh, neat! I wasn't conscious of what you are describing.
So ESLint was right bothering me, even if for me, it wasn't clear why ^^

Did a quick try, and it does address this, indeed, while being more robust, even if the `children[0]` didn't seems to cause side-effects. I always observed singleton in `node[0]`, maybe because `case-police` dictionaries are alphabetical only, without any embedded punctuation in tokens.

thx 👍

**Pull request merged**